exploration/IRI_Zonal.Rmd

```diff
@@ -0,0 +1,177 @@
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+## Intro 
+
+This notebook holds the processing of IRI data to zonal mean and median statistics
+
+
+
+```{r libs_data}
+library(targets)
+library(terra)
+library(tidync)
+library(sf)
+library(tidyverse)
+library(exactextractr)
+library(gghdx)
+library(janitor)
+library(rhdx)
+
+gghdx()
+tar_source()
+
+gdf_aoi_adm <- load_proj_admins() %>% 
+      map(~.x %>% 
+            st_make_valid() %>% 
+            select(matches("^adm\\d_[ep]"))
+      )
+  
+
+fp_iri_prob <- file.path(
+  Sys.getenv("AA_DATA_DIR"),
+  "private",
+  "processed",
+  "lac",
+  "iri",
+  "lac_iri_forecast_seasonal_precipitation_tercile_prob_Np18Sp10Em83Wm93.nc"
+) 
+```
+
+
+
+
+```{r}
+# convert to list of rasters
+lr_iri_prob <- iri_nc_to_r(
+      tnc_object = tidync(fp_iri_prob),
+      type = "prob"
+    ) 
+
+
+# Run zonal stats on each leadtime bavg band 
+df_iri_adm0 <- lr_iri_prob %>% 
+      imap_dfr(
+        \(rtmp,nm){
+          r_bavg <- deepcopy(rtmp[[1]]) # bavg
+          exact_extract(x = r_bavg,
+                        y = gdf_aoi_adm$adm0,
+                        fun=c("mean","median"),
+                        append_cols =c("adm0_es","adm0_pcode")) %>% 
+            pivot_longer(-matches("adm0_")) %>%
+            separate(name, into = c("stat", "date"), sep = "\\.") %>%
+            pivot_wider(names_from = "stat", values_from = "value") %>%
+            
+            # add season information by combing dates w/ leadtimes
+            mutate(
+              leadtime = parse_number(nm),
+              predict_start_mo = as_date(date)+ months(leadtime),
+              predict_mo1_abbr= str_sub(month(predict_start_mo,abbr=T,label=T),start=1,end=1),
+              predict_mo2_abbr= str_sub(month(predict_start_mo+months(1),abbr=T,label=T),start=1,end=1),
+              predict_mo3_abbr= str_sub(month(predict_start_mo+months(2),abbr=T,label=T),start=1,end=1),
+              end_yr_abbr= format(predict_start_mo+months(2),"%y")
+            ) %>% 
+            unite(
+              "seas",predict_mo1_abbr:end_yr_abbr,sep = ""
+            )})
+
+# The above has been integrated into a target
+# it will be accessible in future w/ just
+# tar_load(df_iri_adm0)
+```
+
+
+
+```{r}
+
+# tak max probability (bavg) across leadtimes
+df_iri_adm0_max <- df_iri_adm0 %>% 
+  group_by(
+    across(starts_with("adm")), 
+    predict_start_mo, seas
+  ) %>% 
+  summarise(
+    across(.cols= c("mean","median"), ~max(.x))
+  ) 
+
+
+# filter to just relevant seasons
+df_iri_adm0_max_relevant_season <- df_iri_adm0_max %>% 
+  ungroup() %>% 
+  filter(
+    str_detect(seas,"MJJ|JJA|SON")
+  ) 
+
+```
+
+## Plot Mean Probability
+
+```{r}
+df_iri_adm0_max_relevant_season %>% 
+  ggplot(
+    aes(x= predict_start_mo, y= mean,color=adm0_es)
+  )+
+  geom_point()+
+  geom_line()+
+  scale_x_date(
+    breaks = unique(df_iri_adm0_max_relevant_season$predict_start_mo),
+    labels=unique(df_iri_adm0_max_relevant_season$seas)
```

Comment:
You can put this directly in the `aes()` as `label = seas`.

Reply:
do you mean once the above is implemented we can do that?

The way I currently have it seems to be the only way it has the desired result -if i leave all code and just change the `scale_x_date()` component I get the following:

```
  scale_x_date(
    aes(
    label = seas
    ))
#> returns just years on axis

  scale_x_date(
    aes(
    breaks = predict_start_mo,
    label = seas
    ))
#> same as above

  scale_x_date(
    breaks = unique(df_iri_adm0_max_relevant_season$predict_start_mo),
    aes(
    label = seas
    )
  )
#> correct break placement but labellled w/ yyy-mm-dd instead of labels
```

Reply:
Never mind, was having a fever dream ha!